login-workflow/src/new-architecture/screens/ResetPasswordScreen/ResetPasswordScreen.tsx

```diff
@@ -30,15 +32,28 @@ export const ResetPasswordScreen: React.FC<ResetPasswordScreenProps> = (props) =
     const confirmRef = useRef(null);
     const [passwordInput, setPasswordInput] = useState('');
     const [confirmInput, setConfirmInput] = useState('');
+    const [validCode, setValidCode] = useState(false);
     const [isLoading, setIsLoading] = useState(false);
+    const [errorMsg, setErrorMsg] = useState('');
     const [showSuccessScreen, setShowSuccessScreen] = useState(props.showSuccessScreen);
     const { triggerError, errorManagerConfig } = useErrorManager();
 
-    const { code, email } = useQueryString();
+    const { code, email } = parseQueryString(window.location.search);
 
     const { actions, navigate, routeConfig } = useAuthContext();
     const passwordRequirements = defaultPasswordRequirements(t);
 
+    useEffect(() => {
+        const getVerifyCodeData = async (): Promise<void> => {
+        const verifycode: any = await actions().verifyResetCode(code, email);
```

Comment:
There's no try-catch or .then().catch() handling the error, so if we get an API error it just throws a big red error screen instead of being handled gracefully.